druid-shell/src/platform/mac/appkit.rs

```diff
@@ -73,3 +75,97 @@ impl NSView for id {
         msg_send![self, addTrackingArea: trackingArea]
     }
 }
+
+pub trait NSButton: Sized {
```

Comment:
I don't mind this, but I've also found the trait-based patterns used in the `Cocoa` crate tend not to not feel quite worth it, API-wise, and I tend to just just `msg_send` directly. ymmv.

Reply:
I've found the discoverability of `msg_send` rather tricky, because there's no direct documentation of using the macOS API through it. It requires a bunch of understanding how Objective-C works etc. With a Rust API you can know just Rust.

Reply:
As an example, [here's the Apple docs](https://developer.apple.com/documentation/appkit/nsalert/1524296-beginsheetmodal) for `beginSheetModal(for:completionHandler:)`.

Took me a while to figure out that the declaration given:

```objc
func beginSheetModal(for sheetWindow: NSWindow, 
   completionHandler handler: ((NSApplication.ModalResponse) -> Void)? = nil)
```

Translates into:

```rust
// completionHandler: RcBlock<(NSInteger,), ()>
msg_send![self, beginSheetModalForWindow: window completionHandler: completionHandler]
```

Reply:
a tip about the docs; always switch to the objective-c version, which will match much more closely the code that you end up writing with `msg_send`.